glutin/src/api/egl/display.rs

```diff
@@ -405,9 +414,11 @@ impl Display {
 
     fn get_display(egl: &Egl, display: RawDisplayHandle) -> Result<EglDisplay> {
         let mut display = match display {
-            RawDisplayHandle::Gbm(handle) => handle.gbm_device,
+            RawDisplayHandle::Gbm(handle) => handle.gbm_device.as_ptr(),
             #[cfg(x11_platform)]
-            RawDisplayHandle::Xlib(handle) => handle.display,
+            RawDisplayHandle::Xlib(handle) if handle.display.is_some() => {
+                handle.display.unwrap().as_ptr()
+            },
```

Comment:
It might be cleaner to write this `unwrap()` as an explicit `Some` match:

```suggestion
            RawDisplayHandle::Xlib(XlibDisplayHandle { display: Some(display), .. }) => {
                display.as_ptr()
            },
```